src/Elmish.WPF/WpfProgram.fs

```diff
@@ -11,6 +11,8 @@ type WpfProgram<'model, 'msg> =
     ElmishProgram: Program<unit, 'model, 'msg, unit>
     Bindings: Binding<'model, 'msg> list
     LoggerFactory: ILoggerFactory
+    ErrorHandler: string * exn -> unit
+    MsgErrorHandler: string * exn -> 'msg list
```

Comment:
About just `ErrorHandler: string * exn -> 'msg list` since the variant returning `unit` can be implemented by calling this and returning `[]`?

Reply:
Excellent suggestion. Should we do the same for the two public functions? Just note in the docs that the function may be impure and return an empty list? Or do you prefer keeping them separate? (I may lean very slightly toward keeping them separate, but I don't really have a strong opinion on this.)

Reply:
Let's just have one function.

This introduces a point of fragility, which is that an exception causes this handler to execute, which dispatches a message, which causes an exception, and so on until the application crashes.

You said you had it return `'msg list` because you didn't see a compelling reason not to add it.  Is this reason compelling enough not to add it?  Do you actually want the ability to dispatch a message on an exception?  If not, then I would prefer to avoid point of fragility.

Reply:
> This introduces a point of fragility, which is that an exception causes this handler to execute, which dispatches a message, which causes an exception, and so on until the application crashes.

Good catch. I just verified that the app does indeed enter into an infinite loop.

Note that the effect is the same if we don't allow dispatching messages from the error handler, and the user instead uses e.g. events and subscriptions to dispatch messages on exceptions. That would be the go-to method of getting messages into the loop. There may be valid reasons for handling exceptions in-loop, e.g. modifying the user interface as with any other message (instead of just showing a separate dialog).

I think we should support this, but make it clear that the update handler for the dispatched messages must never throw exceptions, or the app will enter into an infinite loop.

I pushed a commit that is my suggestion. Let me know what you think.

Reply:
If you only intend to use this feature via a function that returns an empty just, then I prefer that the return type be `unit`.  I understand that the user could still get a message dispatched by jumping through more hoops, but this additional fraction is intentional in order to discourage this design.

Reply:
Sounds good to me. YAGNI and all that. Just pushed this change.